contracts/atomic-swap/src/state.rs

```diff
@@ -0,0 +1,92 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{
+    CanonicalAddr, Coin, Env, Order, ReadonlyStorage, StdError, StdResult, Storage,
+};
+use cosmwasm_storage::{bucket, bucket_read, prefixed_read, Bucket, ReadonlyBucket};
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug, Default)]
+pub struct AtomicSwap {
+    /// This is hex-encoded sha-256 hash of the preimage (must be 32*2 = 64 chars)
+    pub hash: String,
```

Comment:
Question - why not just use `Binary` here?

For the public API using hex-encoded string makes sense, but for the internal storage, we can store the hash in base64, which is auto-decoded to `Vec<u8>`